### PR TITLE
Add the combo box for choosing a LUKS version in the container dialog

### PR DESCRIPTION
**Pull request by poncovka**
Users should be able to change the LUKS version of an encrypted container.

Resolves: rhbz#1714120

### CODE REVIEW

**Comment by poncovka** [comment]
Blivet doesn't use the requested LUKS version for container encryption. It is a bug that should be fixed with a pull request https://github.com/storaged-project/blivet/pull/840.

**Comment by poncovka** [comment]
luks_hidden
luks

**Comment by jstodola** [comment]
After changing the version and re-visiting the volume group config window, the version is not updated - it's always luks2.

**Comment by poncovka** [comment]
> After changing the version and re-visiting the volume group config window, the version is not updated - it's always luks2.

@jstodola, have you tested it with https://github.com/storaged-project/blivet/pull/840? Otherwise, Blivet ignores the requested version and uses the default one.

**Comment by jstodola** [comment]
> > After changing the version and re-visiting the volume group config window, the version is not updated - it's always luks2.
> 
> @jstodola, have you tested it with [storaged-project/blivet#840](https://github.com/storaged-project/blivet/pull/840)? Otherwise, Blivet ignores the requested version and uses the default one.

Yes, I put that blivet change into my updates.img, BUT it's very likely that I didn't make the final updates.img correctly. Any chance you have an updates.img I could use for retesting?

File: pyanaconda/modules/storage/partitioning/interactive/utils.py [hunk 1]
```diff
@@ -768,15 +768,24 @@ def generate_device_factory_request(storage, device) -> DeviceFactoryRequest:
...
-        request.container_name = container.name
-        request.container_encrypted = container.encrypted
-        request.container_raid_level = get_device_raid_level_name(container)
-        request.container_size_policy = get_container_size_policy(container)
+        set_container_data(request, container)
 
     return request
 
 
+def set_container_data(request: DeviceFactoryRequest, container):
```
[the remark this answers is not in the document]
Comment (poncovka) [reply pyanaconda/modules/storage/partitioning/interactive/utils.py:792]: The name `set_container_data` is consistent with names of the functions around (`generate_container_data`, `update_container_data`). The container data are considered to be part of the request, so the names are correct.

File: pyanaconda/ui/gui/spokes/lib/custom_storage_helpers.py [hunk 2]
```diff
@@ -625,6 +631,32 @@ def _set_encryption(self):
         if not self._permissions.container_encrypted:
             fancy_set_sensitive(self._encryptCheckbutton, False)
 
+    def _populate_luks(self):
+        """Set up the LUKS version combo box."""
+        # Add the values.
+        self._luks_store.clear()
+        for version in ["luks1", "luks2"]:
+            self._luks_store.append([version])
```
[the remark this answers is not in the document]
Comment (poncovka) [reply pyanaconda/ui/gui/spokes/lib/custom_storage_helpers.py:639]: `GtkListStore` doesn't support it.

File: pyanaconda/ui/gui/spokes/lib/custom_storage_helpers.py [hunk 3]
```diff
@@ -625,6 +631,32 @@ def _set_encryption(self):
...
+
+        # Get the selected value.
+        luks_version = self._request.luks_version or self._device_tree.GetDefaultLUKSVersion()
+
+        # Set the selected value.
+        idx = next(
+            i for i, data in enumerate(self._luks_combo.get_model())
+            if data[0] == luks_version
+        )
```
[the remark this answers is not in the document]
Comment (poncovka) [reply pyanaconda/ui/gui/spokes/lib/custom_storage_helpers.py:648]: We use both ways in GUI. I don't really think that one is better than the other.

**Comment by poncovka** [comment]
@jstodola, I have attached the updates image to the bug. See: https://bugzilla.redhat.com/attachment.cgi?id=1682395

File: pyanaconda/ui/gui/spokes/lib/custom_storage_helpers.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jkonecny12) [reply pyanaconda/ui/gui/spokes/lib/custom_storage_helpers.py:639]: Ahh ok.

**Comment by jstodola** [comment]
The luks version is shown incorrectly even with the updates.img provided by Vendy, so I wonder if it's not related to https://bugzilla.redhat.com/show_bug.cgi?id=1689699 - luks versions behave mysteriously.

**Comment by poncovka** [comment]
@jstodola, it is a bug in Blivet. I will merge this PR and open a new bug for Blivet.

**Comment by poncovka** [comment]
See: https://bugzilla.redhat.com/show_bug.cgi?id=1834373

**Pull request merged**